runtime/vm/stringhelpers.cpp

```diff
@@ -414,39 +414,40 @@ getStringUTF8Length(J9VMThread *vmThread, j9object_t string)
 UDATA 
 verifyQualifiedName(J9VMThread *vmThread, j9object_t string)
 {
+	J9JavaVM* vm = vmThread->javaVM;
 	UDATA unicodeLength = J9VMJAVALANGSTRING_LENGTH(vmThread, string);
-	j9object_t unicodeBytes = J9VMJAVALANGSTRING_VALUE(vmThread, string);
-	UDATA i = 0;
-	UDATA rc = CLASSNAME_VALID_NON_ARRARY;
+	char utf8NameStackBuffer[J9VM_PACKAGE_NAME_BUFFER_LENGTH];
+	J9CfrConstantPoolInfo classNameInfo = {0, 0, 0, 0, 0, NULL, 0};
+	IDATA arity = 0;
+
+	PORT_ACCESS_FROM_JAVAVM(vm);
 
-	if (unicodeLength == 0) {
+	if (0 == unicodeLength) {
 		return CLASSNAME_INVALID;
 	}
 
-	if (IS_STRING_COMPRESSED(vmThread, string)) {
-		if ('[' == J9JAVAARRAYOFBYTE_LOAD(vmThread, unicodeBytes, 0)) {
-			rc = CLASSNAME_VALID_ARRARY;
-		}
-		for (i = 0; i < unicodeLength; i++) {
-			U_8 c = J9JAVAARRAYOFBYTE_LOAD(vmThread, unicodeBytes, i);
-			if (c == '/') {
-				rc = CLASSNAME_INVALID;
-				break;
-			}
-		}
-	} else {
-		if ('[' == J9JAVAARRAYOFCHAR_LOAD(vmThread, unicodeBytes, 0)) {
-			rc = CLASSNAME_VALID_ARRARY;
-		}
-		for (i = 0; i < unicodeLength; i++) {
-			U_16 c = J9JAVAARRAYOFCHAR_LOAD(vmThread, unicodeBytes, i);
-			if (c == '/') {
-				rc = CLASSNAME_INVALID;
-				break;
-			}
-		}
+	classNameInfo.slot1 = (U_32)unicodeLength;
+	classNameInfo.bytes = (U_8*)copyStringToUTF8WithMemAlloc(vmThread, string,
```

Comment:
This needs to be null checked as the allocation can fail in `copyStringToUTF8WithMemAlloc()`

Reply:
already added null check for `classNameInfo.bytes`.